resources/apalache.nix

```diff
@@ -0,0 +1,47 @@
+{
+  apalache-src,
+  pkgs,
+}: let
+  version = "v0.23.1";
+
+  # Patch the build.sbt file so that it does not call the `diff` command,
+  # which is unavailable inside the Nix shell. Note that the diff has a
+  # single space indentation, so auto re-indentation inside the editor
+  # may break the diff.
```

Comment:
sorry, I am confused what is going on here? Why does this need to be called from within the nix shell? You should just be able to run `nix build .#apalache`. But I think I see what you are doing in the patch, apalache gets its version from the local .git and we don't have that here, so this makes sense.

Reply:
It is called during the Nix build. `sbt-derivation` uses `build.sbt` to compute the dependency. But Apalache's `build.sbt` calls `git describe` to dynamically get the version information, and `sbt-derivation` does not include `git` inside the build environment so it fails.

You can try removing the patch and build Apalache again, and it would fail with logs that look like:

```
[info] loading settings for project root from build.sbt ...
[info] resolving key references (11068 settings) ...
java.io.IOException: Cannot run program "git": error=2, No such file or directory
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
        at scala.sys.process.ProcessBuilderImpl$Simple.run(ProcessBuilderImpl.scala:75)
        at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.$bang(ProcessBuilderImpl.scala:119)
        at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.slurp(ProcessBuilderImpl.scala:135)
        at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.$bang$bang(ProcessBuilderImpl.scala:108)
        at $99f7d10e3ffda95b71dd$.$anonfun$tool$6(build.sbt:187)
        at $99f7d10e3ffda95b71dd$.$anonfun$tool$6$adapted(build.sbt:186)
        at scala.Function1.$anonfun$compose$1(Function1.scala:49)
        at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:228)
        at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:170)
        at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
        at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:99)
        at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.IOException: error=2, No such file or directory
        at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
        at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
        at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
        ... 16 more
[error] java.io.IOException: Cannot run program "git": error=2, No such file or directory
[error] Use 'last' for the full log.
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
```

(Note that you might have to slightly modify `depsSha256` to make sure it recomputes the dependencies rather than downloading directly from cache.)

Reply:
Yeah, that's fine, I see why you would get an error from the patch, I just don't understand the comment above.

Reply:
I have clarified a bit inside the comment.

Reply:
Okay, makes sense. For what its worth the issue that it isn't a git repository, I added git as a build input `buildInputs = [pkgs.git]` and got this error:

```
apalache-v0.23.1> fatal: not a git repository (or any of the parent directories): .git
apalache-v0.23.1> java.lang.RuntimeException: Nonzero exit value: 128
apalache-v0.23.1>       at scala.sys.package$.error(package.scala:30)
apalache-v0.23.1>       at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.slurp(ProcessBuilderImpl.scala:138)
apalache-v0.23.1>       at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.$bang$bang(ProcessBuilderImpl.scala:108)
apalache-v0.23.1>       at $db5645a7617e32dfdfc1$.$anonfun$tool$6(build.sbt:187)
apalache-v0.23.1>       at $db5645a7617e32dfdfc1$.$anonfun$tool$6$adapted(build.sbt:186)
apalache-v0.23.1>       at scala.Function1.$anonfun$compose$1(Function1.scala:49)
apalache-v0.23.1>       at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:228)
apalache-v0.23.1>       at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:170)
apalache-v0.23.1>       at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
apalache-v0.23.1>       at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:99)
apalache-v0.23.1>       at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
apalache-v0.23.1>       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
apalache-v0.23.1>       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
apalache-v0.23.1>       at java.base/java.lang.Thread.run(Thread.java:833)
```